### DISCUSSION TITLE
Bulk unique coupon code generation / import / export

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2303] [Ideas]
> Created by: Brian ([brian@knitterain.com](brian@knitterain.com)) on 2016/02/22 17:16:43 +0000
Votes at time of UserVoice import: 9

------
Add the ability for store owners to generate bulk unique coupon codes for product discounts up to 100% off product price, with options for number of codes, prefix, code recipe, max usage, expiration date, and no expiration date. The codes should be unique compared to any existing codes.

Unique code generation and export is perfect for events where store owners can generate redeemable codes in Craft Commerce, export to .csv, and print on business cards, stickers, mail campaigns, email campaigns, etc. for promotional use.

100% off coupon codes work as a redemption code, where consumers can enter this unique code to redeem 1 product at no cost, and not have to enter payment information if the cart is $0.00, but do need to create an account since we sell PDF's where customers need to download periodic updates. (I understand Digital Downloads do not exist at the time of this writing.)

Bulk import is useful for those that have existing methods to generate codes. The file to be imported may need a structure. There should be a warning of conflicting codes to be imported compared to existing codes.

**Comment by lukeholder** [comment]
Split coupons and discounts. Think about this with vouchers codes too.

Once coupon can only belong to one discount.

**Comment by willbocc** [comment]
Do you have any update on this feature becoming available? It would be really useful for an upcoming project to generate bulk unique coupon codes. Thanks

**Comment by mijar** [comment]
I would love to hear about the incorporation of this feature too, our clients request such bulks code usage very often.

**Comment by piotrpog** [comment]
I also have request for that from out clients - any ETA?
